cms/src/hooks/sync-to-legacy/filters/assets.ts

```diff
@@ -0,0 +1,107 @@
+import { oldKnex } from "../oldKnex";
+import { createLocalizable, getStatusFromNew, isObjectUseless, upsertLS } from "../utils";
+import episodes from '../../../btv';
+import { LanguageEntity, VideoEntity } from "@/Database";
+import { ItemsService } from "directus";
+
+enum EncodingVersion {
+    AMS = 0,
+    BTV = 1
+}/* 
+
+function GetEncodingVersion(str): EncodingVersion {
+} */
+
+export async function createAsset(p, m, c) {
+    if (m.collection != "assets") {
+        return
+    }
+    console.log('Asset created!');
+    console.log(p,m,c);
+    // get legacy id
+    p = p as episodes.components["schemas"]["ItemsAssets"]
+    //let image = e.image_file_id as episodes.components["schemas"]["Files"]
+    console.log("directus", p)
+
+    // update it in original        
+    let patch: Partial<VideoEntity> = {
+        EncodingStatus: 2,
+        Filename: p.name,
+        LastUpdate: p.date_created as unknown as Date  ?? new Date(),
+        IsEncrypted: false,
+    }
+    if (p.duration) {
+        var date = new Date(null);
+        date.setSeconds(p.duration);
+        patch.Length = date.toISOString().substr(11, 8);
+    }
+
+    if (p.encoding_version === "btv") {
+        patch.EncodingVersion = EncodingVersion.BTV
+    } else {
+        patch.EncodingVersion = EncodingVersion.AMS
+    }
+
+    console.log("inserting video", patch)
+    let legacyAsset = await oldKnex<VideoEntity>("video").insert(patch).returning("*")
+    console.log("legacyAsset", legacyAsset)
+    p.legacy_id = legacyAsset[0].Id
+    console.log("p", p)
+    return p
```

Comment:
Sometimes you do this and sometimes not. Is there a difference?

Reply:
Do what? I do p.legacy_id = legacyAsset[0].Id on **create**.

Reply:
`return p`

Reply:
it works without it too, so probably doesnt do anything